README.md

```diff
@@ -292,37 +363,73 @@ See: [Compound State](#compound-state) pattern
 
 Do not check existence of `prop` objects.
 
-```coffeescript
-#bad
-render: ->
-  if @props.person?.firstName
-    React.DOM.div(null, @props.person.firstName)
-  else
-    null
-
-#good
-getDefaultProps: ->
-  person:
-    firstName: ''
-
-render: ->
-  React.DOM.div(null, @props.person.firstName)
+```javascript
+// bad
+render() {
+  if (this.props.person) {
+    return <div>{this.props.person.firstName}</div>;
+  } else {
+    return null;
+  }
+}
+```
+
+```javascript
+// good
+getDefaultProps() {
+  return {
+    person: {
+      firstName: 'Guest'
+    }
+  };
+},
+
+render() {
+  return <div>{this.props.person.firstName}</div>;
+}
 ```
 
+This is only where objects or arrays are used. Use PropTypes.shape to clarify
+the types of nested data expected by the component.
+
 **[⬆ back to top](#table-of-contents)**
 
 ## Setting State from Props
 
 Do not set state from props without obvious intent.
 
-```coffeescript
-#bad
-getInitialState: ->
-  items: @props.items
+```javascript
+// bad
+propTypes: {
+  items: React.PropTypes.array
+},
+
+getInitialState() {
+  return {
+    items: this.props.items
+  };
+}
+```
 
-#good
-getInitialState: ->
-  items: @props.initialItems
+```javascript
+// good
+propTypes: {
+  initialItems: React.PropTypes.array
+},
+
+getInitialState() {
+  return {
+    items: this.props.initialItems
+  };
+
+// best (complex)
```

Comment:
Do you want to split `best` from `good` here as well? I'm fine either way but for consistency I think perhaps we should split it.